File: benchmarks/allocchurn/c/allocchurn.c
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char **argv) {
    long n = argc > 1 ? atol(argv[1]) : 5000000;

    long total = 0;
    for (long i = 0; i < n; i++) {
        int *arr = malloc(64 * sizeof(int));
        for (int j = 0; j < 64; j++) arr[j] = j;
        long sum = 0;
        for (int j = 0; j < 64; j++) sum += arr[j];
        total += sum;
        free(arr);
    }

    if (total != n * 2016L) {
        fprintf(stderr, "self-check failed: total mismatch\n");
        return 1;
    }

    printf("%ld\n", total);
    return 0;
}

File: benchmarks/allocchurn/python/allocchurn.py
import sys


def main():
    n = int(sys.argv[1]) if len(sys.argv) > 1 else 5_000_000

    total = 0
    for _ in range(n):
        arr = list(range(64))
        total += sum(arr)

    assert total == n * 2016, "self-check failed: total mismatch"
    print(total)


if __name__ == "__main__":
    main()
